packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -326,16 +351,17 @@ function ComboBoxTray(props: ComboBoxTrayProps) {
 
   React.useEffect(() => {
     focusSafely(inputRef.current);
+  }, []);
 
-    // When the tray unmounts, set state.isFocused (i.e. the tray input's focus tracker) to false.
+  React.useEffect(() => {
+    // When the tray closes, set state.isFocused (i.e. the tray input's focus tracker) to false.
     // This is to prevent state.isFocused from being set to true when the tray closes via tapping on the underlay
     // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
     // Have to do this manually since React doesn't call onBlur when a component is unmounted: https://github.com/facebook/react/issues/12363
-    return () => {
+    if (!state.isOpen && state.isFocused) {
```

Comment:
comment above this line needs correcting, we do this before unmount now

Reply:
hmm I already changed it to say "When the tray closes" instead of "unmounts". Do you mean the line about "React doesn't call onBlur when a component is unmounted"? I think that part is still true?